batcher/message_batcher.go

```diff
@@ -31,8 +43,13 @@ type batcher struct {
 	largestSeq    *big.Int
 	largestSubSeq int
 
+	// smallestSeq and smallestSubSeq are used to track the highest sequence number
+	// of any record in the batch. This is used for checkpointing.
+	smallestSeq    *big.Int
+	smallestSubSeq int
```

Comment:
do you think it would be cleaner if we always passed these around as a single struct?

```
type SeqPair struct {
  SeqNum *big.Int
  SubSeqNum
}
```